### PR TITLE
Fix 20789 - Window's user creation and password policy requirements

### LINKED ISSUE #20789: `user.present` state doesn't show reason for failure when password doesn't match policy requirements
**OS:** `Win 7 Enterprise x86_64 en_US`
**Salt:**

```
               Salt: 2014.7.1
             Python: 2.7.5 (default, May 15 2013, 22:44:16) [MSC v.1500 64 bit (AMD64)]
             Jinja2: 2.7.1
           M2Crypto: 0.21.1
     msgpack-python: 0.4.2
       msgpack-pure: Not Installed
           pycrypto: 2.6
            libnacl: Not Installed
             PyYAML: 3.11
              ioflo: Not Installed
              PyZMQ: 14.1.1
               RAET: Not Installed
                ZMQ: 4.0.4
               Mako: Not Installed
```

**State:**

``` SaltStack
johndoe:
    user.present:
        - password: johndoe
```

**Result:**

```
          ID: johndoe
    Function: user.present
      Result: False
     Comment: Failed to create new user johndoe
     Started: 11:28:05.724000
    Duration: 109.0 ms
     Changes:  
```

**Debug:**

```
    [INFO    ] Running state [johndoe] at time 10:54:56.555000
    [INFO    ] Executing state user.present for johndoe
    [INFO    ] Executing command ['net', 'user', 'johndoe', '/add'] in directory 'C:\\Windows\\system32\\config\\systemprofile'
    [ERROR   ] Command ['net', 'user', 'johndoe', '/add'] failed with return code: 2
    [ERROR   ] stderr: The password does not meet the password policy requirements. Check the minimum password length, password complexity and password history requirements.

    More help is available by typing NET HELPMSG 2245.
    [ERROR   ] retcode: 2
    [ERROR   ] Failed to create new user johndoe
    [INFO    ] Completed state [johndoe] at time 10:54:56.680000

```

**Conclusion:**
The execution module function `win_useradd.add` should handle some common return codes from `net user` (see also: [salt/modules/win_system.py:311](https://github.com/saltstack/salt/blob/2014.7/salt/modules/win_system.py#L311)) and at least log such failures as error.

### PR DESCRIPTION
**Pull request by xeacott**
### What does this PR do?
Surface the errors that occur when user account creation fails on a Window's box.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/20789

### Previous Behavior
```
          ID: johndoe
    Function: user.present
      Result: False
     Comment: Failed to create new user johndoe
     Started: 11:28:05.724000
    Duration: 109.0 ms
     Changes:  


    [INFO    ] Running state [johndoe] at time 10:54:56.555000
    [INFO    ] Executing state user.present for johndoe
    [INFO    ] Executing command ['net', 'user', 'johndoe', '/add'] in directory 'C:\\Windows\\system32\\config\\systemprofile'
    [ERROR   ] Command ['net', 'user', 'johndoe', '/add'] failed with return code: 2
    [ERROR   ] stderr: The password does not meet the password policy requirements. Check the minimum password length, password complexity and password history requirements.

    More help is available by typing NET HELPMSG 2245.
    [ERROR   ] retcode: 2
    [ERROR   ] Failed to create new user johndoe
    [INFO    ] Completed state [johndoe] at time 10:54:56.680000
```

### New Behavior
```
[DEBUG   ] output:
[DEBUG   ] LazyLoaded state.check_result
[DEBUG   ] LazyLoaded highstate.output
local:
----------
          ID: create_user
    Function: user.present
        Name: testuser123
      Result: False
     Comment: The password does not meet the password policy requirements. Check the minimum password length, password complexity and password history requirements.
     Started: 16:00:47.406333
    Duration: 99.254 ms
     Changes:

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:  99.254 ms
```

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/win_useradd.py [hunk 1]
```diff
@@ -255,7 +251,7 @@ def update(
     .. code-block:: bash
 
         salt '*' user.update bob password=secret profile=C:\\Users\\Bob
-                 home=\\server\homeshare\bob homedrive=U:
+                 home=\\server\\homeshare\bob homedrive=U:
```
Comment (twangboy) [inline salt/modules/win_useradd.py:254]: Probably need another slash before bob as well

File: salt/modules/win_useradd.py [hunk 2]
```diff
@@ -175,7 +171,7 @@ def add(
         log.error("nbr: %s", exc.winerror)
         log.error("ctx: %s", exc.funcname)
         log.error("msg: %s", exc.strerror)
-        return False
+        return exc.strerror
```
Comment (twangboy) [inline salt/modules/win_useradd.py:174]: Let's raise a CommandExecutionError here instead:
```
raise CommandExecutionError(exc.strerror)
```

**Review by twangboy — CHANGES_REQUESTED** [review]
Let's raise a CommandExecutionError when the command fails

File: salt/states/user.py [hunk 3]
```diff
@@ -854,8 +847,13 @@ def _change_homedir(name, val):
                 "logonscript": win_logonscript,
             }
 
-        if __salt__["user.add"](**params):
-            ret["comment"] = "New user {0} created".format(name)
+        result = __salt__["user.add"](**params)
+        if result is not True and salt.utils.platform.is_windows():
```
Comment (twangboy) [inline salt/states/user.py:851]: Why is this Windows specific?
If it is Windows specific, what happens when it is `not True` but not Windows?

**Review by twangboy — CHANGES_REQUESTED** [review]

**Review by twangboy — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Review by krionbsd — APPROVED** [review]

**Pull request merged**